apps/projection/src/app/app.component.ts

```diff
@@ -1,18 +1,45 @@
 import { Component } from '@angular/core';
-import { CityCardComponent } from './component/city-card/city-card.component';
-import { StudentCardComponent } from './component/student-card/student-card.component';
-import { TeacherCardComponent } from './component/teacher-card/teacher-card.component';
+import { CardComponent } from './ui/card/card.component';
+import { CardType } from './model/card.model';
+import { NgFor, NgIf } from '@angular/common';
 
 @Component({
   selector: 'app-root',
   template: `
     <div class="grid grid-cols-3 gap-3">
-      <app-teacher-card></app-teacher-card>
-      <app-student-card></app-student-card>
-      <app-city-card></app-city-card>
+      <app-card
+        *ngFor="let card of cards"
+        [cardType]="card.type"
+        [labelKey]="card.labelKey"
+        [style.--bg-color]="card.bgColor || '#FFFFFF'">
+        <ng-template #image [ngIf]="card.imageSrc">
+          <img [src]="card.imageSrc" width="200px" />
+        </ng-template>
+      </app-card>
     </div>
   `,
   standalone: true,
-  imports: [TeacherCardComponent, StudentCardComponent, CityCardComponent],
+  imports: [NgIf, NgFor, CardComponent],
 })
-export class AppComponent {}
+export class AppComponent {
+  types = [CardType.TEACHER, CardType.STUDENT, CardType.CITY];
+
+  cards = [
+    {
+      type: CardType.TEACHER,
+      labelKey: 'firstname',
+      imageSrc: 'assets/img/teacher.png',
+      bgColor: 'rgba(250, 0, 0, 0.1)',
+    },
+    {
+      type: CardType.STUDENT,
+      labelKey: 'firstname',
+      imageSrc: 'assets/img/student.webp',
+      bgColor: 'rgba(0, 250, 0, 0.1)',
+    },
+    {
+      type: CardType.CITY,
+      labelKey: 'name',
+    },
+  ];
```

Comment:
I'm not a big fan of this solution with static array of paramater. Not very easy to customize and all element must have the same shape. And you need to pass a type to know which store to hit.
Since Angular is a component based framework, it's always better to separate each element inside its own component. easier to customize at will. If you need to change one thing inside one element, you don't need to add if statement every where and you only need to impact one component.